### ISSUE TITLE
NPE with 23.2.0 in android.support.v7.widget.RecyclerView$ItemAnimator$ItemHolderInfo.left

### ISSUE THREAD

**Issue by gabrielemariotti** [gabrielemariotti/changeloglib#31]
As reported by Zeliret in #28:

For me there is another problem. I have a crash: 

```java
java.lang.NullPointerException: Attempt to read from field 'int android.support.v7.widget.RecyclerView$ItemAnimator$ItemHolderInfo.left' on a null object reference
                                                                 at android.support.v7.widget.SimpleItemAnimator.animateDisappearance(SimpleItemAnimator.java:78)
                                                                 at android.support.v7.widget.RecyclerView.animateDisappearance(RecyclerView.java:3246)
                                                                 at android.support.v7.widget.RecyclerView.access$700(RecyclerView.java:147)
                                                                 at android.support.v7.widget.RecyclerView$4.processDisappeared(RecyclerView.java:422)
                                                                 at android.support.v7.widget.ViewInfoStore.process(ViewInfoStore.java:231)
                                                                 at android.support.v7.widget.RecyclerView.dispatchLayoutStep3(RecyclerView.java:3086)
                                                                 at android.support.v7.widget.RecyclerView.dispatchLayout(RecyclerView.java:2914)
                                                                 at android.support.v7.widget.RecyclerView.onLayout(RecyclerView.java:3277)
                                                                 at android.view.View.layout(View.java:16636)
                                                                 at android.view.ViewGroup.layout(ViewGroup.java:5437)
                                                                 at android.widget.FrameLayout.layoutChildren(FrameLayout.java:336)
                                                                 at android.widget.FrameLayout.onLayout(FrameLayout.java:273)
                                                                 at android.view.View.layout(View.java:16636)
                                                                 at android.view.ViewGroup.layout(ViewGroup.java:5437)
                                                                 at android.widget.FrameLayout.layoutChildren(FrameLayout.java:336)
                                                                 at android.widget.FrameLayout.onLayout(FrameLayout.java:273)
                                                                 at android.view.View.layout(View.java:16636)
                                                                 at android.view.ViewGroup.layout(ViewGroup.java:5437)
                                                                 at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1743)
                                                                 at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1586)
                                                                 at android.widget.LinearLayout.onLayout(LinearLayout.java:1495)
                                                                 at android.view.View.layout(View.java:16636)
                                                                 at android.view.ViewGroup.layout(ViewGroup.java:5437)
                                                                 at android.widget.FrameLayout.layoutChildren(FrameLayout.java:336)
                                                                 at android.widget.FrameLayout.onLayout(FrameLayout.java:273)
                                                                 at android.view.View.layout(View.java:16636)
                                                                 at android.view.ViewGroup.layout(ViewGroup.java:5437)
                                                                 at android.widget.FrameLayout.layoutChildren(FrameLayout.java:336)
                                                                 at android.widget.FrameLayout.onLayout(FrameLayout.java:273)
                                                                 at android.view.View.layout(View.java:16636)
                                                                 at android.view.ViewGroup.layout(ViewGroup.java:5437)
                                                                 at android.widget.FrameLayout.layoutChildren(FrameLayout.java:336)
                                                                 at android.widget.FrameLayout.onLayout(FrameLayout.java:273)
                                                                 at android.view.View.layout(View.java:16636)
                                                                 at android.view.ViewGroup.layout(ViewGroup.java:5437)
                                                                 at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1743)
                                                                 at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1586)
                                                                 at android.widget.LinearLayout.onLayout(LinearLayout.java:1495)
                                                                 at android.view.View.layout(View.java:16636)
                                                                 at android.view.ViewGroup.layout(ViewGroup.java:5437)
                                                                 at android.widget.FrameLayout.layoutChildren(FrameLayout.java:336)
                                                                 at android.widget.FrameLayout.onLayout(FrameLayout.java:273)
                                                                 at com.android.internal.policy.PhoneWindow$DecorView.onLayout(PhoneWindow.java:2678)
                                                                 at android.view.View.layout(View.java:16636)
                                                                 at android.view.ViewGroup.layout(ViewGroup.java:5437)
                                                                 at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:2171)
                                                                 at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:1931)
                                                                 at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1107)
                                                                 at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:6013)
                                                                 at android.view.Choreographer$CallbackRecord.run(Choreographer.java:858)
                                                                 at android.view.Choreographer.doCallbacks(Choreographer.java:670)
                                                                 at android.view.Choreographer.doFrame(Choreographer.java:606)
                                                                 at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:844)
                                                                 at android.os.Handler.handleCallback(Handler.java:739)
                                                                 at android.os.Handler.dispatchMessage(Handler.java:95)
                                                                 at android.os.Looper.loop(Looper.java:148)
                                                                 at android.app.ActivityThread.main(ActivityThread.java:5417)
                                                                 at java.lang.reflect.Method.invoke(Native Method)
                                                                 at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
                                                             	at com.android.internal.os.ZygoteInit.main(ZygoteInit.jav
```

My layout:
```xml
<it.gmariotti.changelibs.library.view.ChangeLogRecyclerView
    android:id="@+id/view"
    xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:app="http://schemas.android.com/apk/res-auto"
    android:layout_width="match_parent"
    android:layout_height="wrap_content"
    android:clipToPadding="false"
    android:paddingBottom="@dimen/padding_narrow"
    android:paddingTop="@dimen/padding_narrow"
    app:rowHeaderLayoutId="@layout/header_changelog"
    app:rowLayoutId="@layout/item_changelog"/>
```

**Comment by gabrielemariotti** [comment]
I am investigating about it. It seems to happen only in same cases as a Dialog. 
Currently I am able to replicate the bug but still not able to fix it.

It seems related to the RecyclerView v23.2.0 and should be addressed by the Issue on[ google tracker ](https://code.google.com/p/android/issues/detail?id=193958&colspec=ID%20Type%20Status%20Owner%20Summary%20Stars#makechanges).
Check the comment 27 and below.

**Comment by Zeliret** [comment]
Good news. 23.2.1 fixes this crash. ;D

**Comment by NitroXenon** [comment]
After I updated to 23.3.0, the dialog change log became weird. There is a large space between each item.

**Comment by gabrielemariotti** [comment]
Fixed with the 2.1.0

**Comment by denniskorir** [comment]
tcexxygcuijjhjng y dj,ij zqg yn,CDu u dc
